From a70d2836d6829ef0e876e2f66a312e9a69686e5f Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Wed, 18 Oct 2017 10:43:20 +0800
Subject: [PATCH 2651/5242] MLK-16600 gpu: imx: dpu: common: Initialize pixel
 link in resume() only if necessary

commit  e7c9cc0c466a68e3bb5c24ecb2364b4268fa0e32 from
https://source.codeaurora.org/external/imx/linux-imx.git

We should initialize pixel link in resume() for DPUv2 which
has pixel link quirks, but not for DPUv1 which hasn't the quirks.

Fixes: 0d7fa2aa1a9f ("MLK-16581-6 gpu: imx: dpu: Add system power management support")
Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu/dpu-common.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/imx/dpu/dpu-common.c b/drivers/gpu/imx/dpu/dpu-common.c
index b14ce54..9f00dd1 100644
--- a/drivers/gpu/imx/dpu/dpu-common.c
+++ b/drivers/gpu/imx/dpu/dpu-common.c
@@ -1560,7 +1560,8 @@ static int dpu_resume(struct device *dev)
 
 	dpu_intsteer_enable_lines(dpu);
 
-	dpu_pixel_link_init(dpu->id);
+	if (dpu->devtype->pixel_link_quirks)
+		dpu_pixel_link_init(dpu->id);
 
 	_dpu_submodules_init(dpu, pdev);
 
-- 
1.7.9.5

